From 39d5a9f69c583f660221a1085f2495fa140fae31 Mon Sep 17 00:00:00 2001
From: Terje Larsen <terlar@gmail.com>
Date: Thu, 23 Aug 2018 22:33:44 +0200
Subject: [PATCH] Support using a relative path

Sometimes it is not feasible to use the full path, either when using
bin-stub wrappers going to docker images and other cases. This provides
an option to opt-out from the absolute paths.
---
 rspec-mode.el | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rspec-mode.el b/rspec-mode.el
index a7de2ca..46ccc95 100644
--- a/rspec-mode.el
+++ b/rspec-mode.el
@@ -178,6 +178,11 @@
   :type 'string
   :group 'rspec-mode)
 
+(defcustom rspec-use-relative-path nil
+  "When t the file path will be relative."
+  :type 'boolean
+  :group 'rspec-mode)
+
 (defcustom rspec-use-bundler-when-possible t
   "When t and Gemfile is present, run specs with 'bundle exec'.
 Not used when running specs using Zeus or Spring."
@@ -705,6 +710,7 @@ file if it exists, or sensible defaults otherwise."
                                          rspec-docker-cwd file))
       (vagrant (replace-regexp-in-string (regexp-quote (rspec-project-root))
                                          rspec-vagrant-cwd file))
+      (rspec-use-relative-path (file-relative-name file (rspec-project-root)))
       (t  file)))))
 
 (defun rspec--docker-wrapper (command)
